coffea/processor/executor.py

```diff
@@ -1071,11 +1071,6 @@ def _preprocess_fileset(self, fileset: Dict) -> None:
                     filemeta.maybe_populate(self.metadata_cache)
         else:
             for filemeta in fileset:
-                # not sure why need to check for bad files here... otherwise pop fails below with pytest.
-                # fmt: off
-                if self.skipbadfiles and not filemeta.populated(clusters=self.align_clusters):  # noqa  # fmt: skip <-- gets ignored, see: https://github.com/psf/black/issues/2421
```

Comment:
I guess this is no longer failing? It seems `_filter_badfiles` does this job later anyway?
Maybe #375 is fixed now as well?

Reply:
It goes a little deeper than this. Fixing issues now.

Reply:
Basically, there shouldn't be a if-block here for skipbadfiles.

Reply:
Doesn't this make the pre-executor open all files even if only one chunk is desired then? The point of special handling when maxchunks=True was originally to speed up testing on one file #86

Reply:
Well, we were doing it in a funny way if that was the case. This is making it more clear we want to be able to pass in pickled or otherwise cached preprocessing.